lib/deref.js

```diff
@@ -302,8 +302,8 @@ module.exports = {
         schema.type = 'string';
       }
 
-      // Discard format if not supported by both json-schema-faker and ajv
-      if (!_.includes(SUPPORTED_FORMATS, schema.format)) {
+      // Discard format if not supported by both json-schema-faker and ajv or pattern is also defined
+      if (!_.includes(SUPPORTED_FORMATS, schema.format) || (_.has(schema, 'pattern') && _.has(schema, 'format'))) {
```

Comment:
why not just (schema.pattern && schema.format)? these will always be truthy if defined, right?

Reply:
@abhijitkane Yes, it would be truthy. Will change it.